### ISSUE TITLE
absorption and reflectance from dielectric constants

### ISSUE THREAD

**Issue by computron** [materialsintelligence/propnet#8]
reflectance should look like [(1-n)^2+k^2]/[(1+n)^2+k^2] nd you can remove k for very transparent materials.

absoprtion should look like 4*pi*k/lambda where lambda is wavelength of light

Also add the relationship between  n and k and eps_real and eps_imag, should be straightforward. Thus if someone starts with eps_real and eps_imag, they can end up with reflectance / absorptivity

**Comment by computron** [comment]
you can also get transmittance if you know the sample thickness through

A + R + T = 1

where R is the reflectance above
A is the absorption coefficient above times sample thickness

**Comment by vtshitoyan** [comment]
Just clarifying things for myself:

n - refractive index
k - extinction coefficient
N = n + j*k - complex refraction index
\alpha = 4\pi{k}/\lambda = 2\omega{k}/c - absorption coefficient
R = [(1-n)^2+k^2]/[(1+n)^2+k^2] - Reflectance
A = \alpha * thickness - Absorption
A + R + T - 1 = 0

https://unlcms.unl.edu/cas/physics/tsymbal/teaching/SSP-927/Section%2013_Optical_Properties_of_Solids.pdf

**Comment by computron** [comment]
yes and also the relationship between [n, k] and [eps1, eps2] - let me know if you can't find it

**Comment by vtshitoyan** [comment]
@computron is the formula for getting Transmittance correct given the definitions? If my understanding is correct

I_0 = I(1-R) would be the intensity that passes into the material
I_t = I(1-R)exp(-A) = IT would be the intensity that comes out
Therefore T = (1-R)exp(-A)

In order to get 
1 = A' + R + T
I have to define A' = (1-R)(1 - exp(-A)), where A' would correspond to the amount of absorbed intensity (absorbance?).

Is the naming below correct?
T - transmittance
R - reflectance
A/A' - absorption/absorbance (?)

**Comment by computron** [comment]
I am typically more interested in intrinsic properties of the material (e.g. reflectivity, transmittivity) than extrinsic properties (e.g., reflectance, transmittance). The latter require knowing things like the thickness of the sample whereas the former just requires knowing the materials properties.

As a start I'd want the parameters that are represented in the following code:

https://github.com/bryanhoffman/plotoptics

This includes

- computing n and k from e1 and e2, e.g. see https://github.com/bryanhoffman/plotoptics/blob/master/plot_reflection.R

```
k <- sqrt((sqrt(e1^2+e2^2)-e1)/2)
n <- sqrt((sqrt(e1^2+e2^2)+e1)/2)
```

Getting the:

- absorption coefficient: https://github.com/bryanhoffman/plotoptics/blob/master/plot.R or https://github.com/bryanhoffman/plotoptics/blob/master/plot_lambda.R depending on whether you are doing it as a function of energy or lambda
- loss function (related to transmittance): https://github.com/bryanhoffman/plotoptics/blob/master/plot_loss.R
- reflectivity: https://github.com/bryanhoffman/plotoptics/blob/master/plot_reflection.R

There is also tauc plot in the above repo but IMO that one is not important

**Comment by mkhorton** [comment]
Should this issue be closed, or are there any more related ones to add?

**Comment by vtshitoyan** [comment]
I think one can improve the models to include magnetic permeability in a more consistent manner, and I also wanted to think about frequency dependencies in more detail. Maybe we can leave this open for now?

**Comment by mkhorton** [comment]
Sure, sounds good :)
